### PR TITLE
Update pyyaml to 5.4 to address security vulnerabilities

### PR DESCRIPTION
**Pull request by smcd253**
Vulnerability Fix:
Update: pyyaml==5.4

The following dependencies were updated to accommodate this fix:
docker-compose==1.28.5
docker[ssh]>=4.4.4

Addressing issue #303 

Vulnerability Descriptions:
1) A vulnerability was discovered in the PyYAML library in versions before 5.3.1, where it is susceptible to arbitrary code execution when it processes untrusted YAML files through the full_load method or with the FullLoader loader. Applications that use the library to process untrusted input may be vulnerable to this flaw. An attacker could use this flaw to execute arbitrary code on the system by abusing the python/object/new constructor.

2) A vulnerability was discovered in the PyYAML library in versions before 5.4, where it is susceptible to arbitrary code execution when it processes untrusted YAML files through the full_load method or with the FullLoader loader. Applications that use the library to process untrusted input may be vulnerable to this flaw. This flaw allows an attacker to execute arbitrary code on the system by abusing the python/object/new constructor. This flaw is due to an incomplete fix for CVE-2020-1747.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 autopep8
 backports.unittest-mock
 click
-docker[ssh]>=3.6.0,<4.0
```
Comment (marianan) [inline requirements.txt:4]: >4.0 [](start = 20, length = 3)

do you know why was docker 4.0 an issue before?

File: requirements.txt [hunk 2]
```diff
@@ -11,9 +11,9 @@ applicationinsights
 prompt_toolkit
 rope
 tox
-pyyaml>=4.1,<=4.2b4
+pyyaml==5.4
```
Comment (marianan) [inline requirements.txt:14]: >== [](start = 6, length = 2)

why == and not >=?

File: requirements.txt [hunk 3]
```diff
@@ -11,9 +11,9 @@ applicationinsights
...
 jsonpath_rw
-docker-compose>=1.21.0,<=1.24.0
+docker-compose==1.28.5
```
Comment (marianan) [inline requirements.txt:16]: >== [](start = 14, length = 2)

why fixing on this particular version and not this one or higher?

**Review by marianan — CHANGES_REQUESTED** [review]
:clock1:

File: requirements.txt (hunk 1 above)
Comment (smcd253) [reply requirements.txt:4]: it was not. this version was a dependent on pyyaml 4.1. docker 4.4.4 and greater works with pyyaml 5.4

File: requirements.txt (hunk 2 above)
Comment (smcd253) [reply requirements.txt:14]: didn't want to extrapolate beyond the requirements

File: requirements.txt (hunk 3 above)
Comment (smcd253) [reply requirements.txt:16]: same as above. didn't want to extrapolate beyond the component governance alert recommendations

**Review by marianan — APPROVED** [review]
:shipit:

**Pull request merged**